### ISSUE TITLE
It would be nice to use this plugin with ui entities

### ISSUE THREAD

**Issue by vav-dev** [Weasy666/bevy_svg#15]
It would be nice if Svg2dBundle could be used with ui entities like this:
```rust
container
    .spawn_bundle(ButtonBundle {
        ..default()
    })
    .with_children(|parent| {
        parent.spawn_bundle(Svg2dBundle {
            svg: assets_resource.cogwheel.clone(),
            origin: Origin::Center,
            ..default()
        });
    });
```

**Comment by Weasy666** [comment]
Yeah, that sounds like a nice idea! I would have to check what needs to be done to make this possible.
Have you tried it? Did you get any error, or did it just not work?

**Comment by vav-dev** [comment]
It crashes with an error in the console:
```
Styled child in a non-UI entity hierarchy. You are using an entity with UI components as a child of an entity without UI components, results may be unexpected
```

**Comment by cornytrace** [comment]
I'm facing a similar problem myself. To make this work you either need to do what ImageBundle does to make a Handle<Image> render on top of the UI but for Mesh2dHandle instead, or you need to render the mesh2d to an Image and use that in the ImageBundle. Both are sufficiently general operations that I hope functionality for this can be added to bevy itself. 
A third option can be to make an SvgTextureLoader which takes a path and a size and directly creates an image. Though we probably want a way to rerender the svg if the scale changes, and I don't know how possible or complicated that is.
